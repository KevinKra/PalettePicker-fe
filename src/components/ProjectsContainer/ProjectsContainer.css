.projects-container-header {
  display: flex;
  justify-content: center;
  border-right: 1px dashed coral;
  border-left: 1px dashed coral;
  min-height: 30vh;
  color: #ffffff;
}

.projects-container-header > p {
  padding-top: 8.5vh;
}
