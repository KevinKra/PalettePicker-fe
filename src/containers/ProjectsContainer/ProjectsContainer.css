.projects-container-header {
  display: flex;
  justify-content: center;
  border-right: 1px solid coral;
  border-left: 1px solid coral;
  color: #ffffff;
  transition: 0.5s height ease-in;
}

.projects-container-header > p {
  transition: 1s padding-top ease-in;
  padding-top: 2.5vh;
}

.header-hidden {
  height: 8vh;
}

.header-hidden > p {
  padding-top: 2vh;
}
