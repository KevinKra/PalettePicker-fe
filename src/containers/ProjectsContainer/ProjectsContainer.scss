@import "../../styles/main.scss";

.ProjectsContainer {
	grid-row: 2 / 3;
	transition: 1s min-height ease-in-out;
	.projects-header {
		display: flex;
		justify-content: center;
		align-items: center;
		height: 100%;
		color: $color-primary;
	}
}
