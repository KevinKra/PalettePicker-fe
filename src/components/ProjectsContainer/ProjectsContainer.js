import React, { Component } from "react";
import Project from "../Project/Project";
import "./ProjectsContainer.css";

export default class ProjectsContainer extends Component {
  state = {
    user: { userId: "123ddEad2199E", username: "Elliot" },
    projects: [
      {
        projectTitle: "--Send to Steve",
        projectId: "asd3213ey1u",
        projectDesc: "Palettes that Steve requested ...",
        palettes: [
          {
            paletteTitle: "Ok here is title",
            paletteColors: ["#45CD12", "#DFD442", "#4E3211"]
          },
          { paletteTitle: "Idea - 2 Colors", paletteColors: ["#123456"] },
          {
            paletteTitle: "Serene",
            paletteColors: ["#FEDCBA", "#45CD12", "#3A3A3A", "#123123"]
          },
          {
            paletteTitle: "-- Send to Steve",
            paletteColors: [
              "#FEDCBA",
              "#FEDFED",
              "#ABCABC",
              "#123123",
              "#3A3A3A"
            ]
          }
        ]
      },
      {
        projectTitle: "My Favorites!",
        projectId: "3ykuyOP23eeE",
        projectDesc: "A list of my favorite palettes.",
        palettes: [
          {
            paletteTitle: "For Mary",
            paletteColors: ["#568111", "#512612", "#EA31EA"]
          },
          { paletteTitle: "Idea - ???", paletteColors: ["#123456", "#99EE32"] },
          {
            paletteTitle: "Iteration 2",
            paletteColors: ["#FEDCBA", "#FFFFDD", "#AABBE1", "#3A3A3A"]
          }
        ]
      }
    ]
  };
  render() {
    const projects = this.state.projects.map(project => {
      return (
        <Project title={project.projectTitle} palettes={project.palettes} />
      );
    });
    const textMessage = this.state.projects.length ? (
      <p>Here are your projects!</p>
    ) : (
      <p>This area is where your projects will be saved!</p>
    );

    return (
      <div>
        <section className="ProjectsContainer">
          <React.Fragment>
            <header
              className={`projects-container-header ${
                this.state.projects.length ? null : "header-hidden"
              }`}
            >
              {textMessage}
            </header>
            {projects}
          </React.Fragment>
        </section>
      </div>
    );
  }
}
