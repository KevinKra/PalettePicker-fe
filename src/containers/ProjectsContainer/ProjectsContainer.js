import React, { Component } from "react";
import Project from "../../components/Project/Project";
import * as actions from "../../_redux/actions/index";
import { connect } from "react-redux";
import "./ProjectsContainer.css";

class ProjectsContainer extends Component {
  state = {
    user: { userId: "123ddEad2199E", username: "Elliot" }
  };

  componentDidMount() {
    this.props.addUserProjects([
      {
        projectTitle: "--Send to Steve",
        projectId: "asd3213ey1u",
        projectDesc: "Palettes that Steve requested ...",
        palettes: [
          {
            paletteTitle: "Ok here is title",
            paletteColors: ["#45CD12", "#DFD442", "#4E3211"],
            paletteId: "asd3a1DD3sd"
          },
          {
            paletteTitle: "Idea - 2 Colors",
            paletteColors: ["#123456"],
            paletteId: "eeEE123D3sd"
          },
          {
            paletteTitle: "Serene",
            paletteColors: ["#FEDCBA", "#45CD12", "#3A3A3A", "#123123"],
            paletteId: "f321DFDff3sd"
          },
          {
            paletteTitle: "-- Send to Steve",
            paletteColors: [
              "#FEDCBA",
              "#FEDFED",
              "#ABCABC",
              "#123123",
              "#3A3A3A"
            ],
            paletteId: "eee23aOO3sd"
          },
          {
            paletteTitle: "Iteration 4",
            paletteColors: [
              "#FEDCBA",
              "#FFFFDD",
              "#45CD12",
              "#AABBE1",
              "#3A3A3A"
            ],
            paletteId: "13ASeeU98MN"
          },
          {
            paletteTitle: "- Two Potential Options -",
            paletteColors: ["#AFFFDD", "#1A3A3A"],
            paletteId: "13AS3U98MN"
          }
        ]
      },
      {
        projectTitle: "My Favorites!",
        projectId: "3ykuyOP23eeE",
        projectDesc: "A list of my favorite palettes.",
        palettes: [
          {
            paletteTitle: "For Mary",
            paletteColors: ["#568111", "#512612", "#EA31EA"],
            paletteId: "oi112uTR3y"
          },
          {
            paletteTitle: "Idea - ???",
            paletteColors: ["#123456", "#99EE32"],
            paletteId: "833AS3U99M"
          },
          {
            paletteTitle: "Iteration 2",
            paletteColors: ["#FEDCBA", "#FFFFDD", "#AABBE1", "#3A3A3A"],
            paletteId: "13AS3U98MN"
          }
        ]
      }
    ]);
  }

  render() {
    const projects = this.props.userProjects.map(project => {
      return (
        <Project
          title={project.projectTitle}
          palettes={project.palettes}
          key={project.projectId}
        />
      );
    });
    const textMessage = this.props.userProjects.length ? (
      <p>Here are your projects!</p>
    ) : (
      <p>This area is where your projects will be saved!</p>
    );

    return (
      <div>
        <section className="ProjectsContainer">
          <React.Fragment>
            <header
              className={`projects-container-header ${
                this.props.userProjects.length ? null : "header-hidden"
              }`}
            >
              {textMessage}
            </header>
            {projects}
          </React.Fragment>
        </section>
      </div>
    );
  }
}

const mapStateToProps = store => ({
  userProjects: store.userProjects
});

const mapDispatchToProps = dispatch => ({
  addUserProjects: projects => dispatch(actions.addUserProjects(projects))
});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(ProjectsContainer);
