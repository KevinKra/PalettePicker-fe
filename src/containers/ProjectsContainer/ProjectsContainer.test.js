import React from "react";
import { shallow } from "enzyme";
import { ProjectsContainer } from "./ProjectsContainer";

describe("ProjectsContainer", () => {
  it("matches snapshot", () => {
    const props = {
      userProjects: [
        {
          projectTitle: "t",
          palettes: [],
          projectId: 9,
          removeProject: jest.fn(),
          removePalette: jest.fn()
        }
      ]
    };
    const wrapper = shallow(<ProjectsContainer {...props} />, {
      disableLifecycleMethods: true
    });
    expect(wrapper).toMatchSnapshot();
  });
});
