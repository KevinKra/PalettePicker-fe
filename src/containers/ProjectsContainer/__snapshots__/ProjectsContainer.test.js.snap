// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProjectsContainer matches snapshot 1`] = `
<section
  className="ProjectsContainer"
>
  <header
    className="projects-container-header"
  >
    <p>
      Here are your projects!
    </p>
  </header>
  <TransitionGroup
    childFactory={[Function]}
    component="div"
  >
    <Project
      id={9}
      index={0}
      key="9"
      palettes={Array []}
      project={
        Object {
          "palettes": Array [],
          "projectId": 9,
          "projectTitle": "t",
          "removePalette": [MockFunction],
          "removeProject": [MockFunction],
        }
      }
      removePalette={[Function]}
      removeProject={[Function]}
      title="t"
    />
  </TransitionGroup>
</section>
`;
